country-a-service/epps-service/src/test/java/dk/nsp/epps/service/mapping/DispensationMapperTest.java

```diff
@@ -42,89 +58,110 @@ StartEffectuationResponseType testStartEffectuationResponse(Document cda) throws
-        Assertions.assertFalse(org.getTelephoneNumber().isBlank());
     }
 
-    @Test
-    void startEffectuationRequestPrescriptionTest() throws XPathExpressionException, MapperException {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void startEffectuationRequestPrescriptionTest(String xmlFileName, String mappingErrorMessageExpected) throws XPathExpressionException, MapperException {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
         var p = sut.startEffectuationRequestPrescription(cda);
 
         Assertions.assertTrue(p.getIdentifier() > 0L);
     }
 
-    @Test
-    void effectuationTest() throws Exception {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void effectuationTest(String xmlFileName, String mappingErrorMessageExpected) throws Exception {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
+        if (mappingErrorMessageExpected != null) {
+            var exception = Assertions.assertThrows(DataRequirementException.class, () -> testStartEffectuationResponse(cda));
+            Assertions.assertEquals(exception.getMessage(), mappingErrorMessageExpected);
+            return;
+        }
+
         var startEffectuationResponse = testStartEffectuationResponse(cda);
 
         var e = sut.effectuation(cda, startEffectuationResponse);
 
         Assertions.assertTrue(e.getDateTime().isValid());
         var packageDispensed = e.getPackageDispensed();
-        Assertions.assertTrue(packageDispensed.getPackageQuantity()>0);
+        Assertions.assertTrue(packageDispensed.getPackageQuantity() > 0);
         assertThat(packageDispensed.getPackageNumber().getSource(), is(equalTo("Medicinpriser")));
         Assertions.assertFalse(packageDispensed.getPackageNumber().getValue().isBlank());
+
     }
 
-    @Test
-    void packageQuantityTest() throws Exception {
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void packageQuantityTest(String xmlFileName, String mappingErrorMessageExpected) throws Exception {
         var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+        var cda = testDispensationCda(xmlFileName);
 
         var q = sut.packageQuantity(cda);
 
         Assertions.assertTrue(q > 0);
     }
 
-    @Test
-    void createPharmacyEffectuationRequestTest() throws MapperException, XPathExpressionException {
-        var sut = new DispensationMapper();
-        var cda = testDispensationCda();
+    @ParameterizedTest
+    @MethodSource("testDispensationCdas")
+    void createPharmacyEffectuationRequestTest(String xmlFileName, String mappingErrorMessageExpected) throws MapperException, XPathExpressionException {
+        var sut = new DispensationMapper(); //TODO why is this called sut?
```

Comment:
sut = "System Under Test".  This is the DispensationMapper test after all 🙂

Reply:
https://jabde.com/2022/03/13/three-letter-acronym-crisis/